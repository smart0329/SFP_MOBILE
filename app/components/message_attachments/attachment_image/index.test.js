// Copyright (c) 2019-present Smart.Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {Image} from 'react-native';
import {shallow} from 'enzyme';
import React from 'react';

const originalGetSizeFn = Image.getSize;

import Preferences from 'mattermost-redux/constants/preferences';

import AttachmentImage from './index';

describe('AttachmentImage', () => {
    const baseProps = {
        deviceHeight: 256,
        deviceWidth: 128,
        imageMetadata: {width: 32, height: 32},
        imageUrl: 'https://images.com/image.png',
        theme: Preferences.THEMES.default,
    };

    afterEach(() => {
        Image.getSize = originalGetSizeFn;
    });

    test('it matches snapshot', () => {
        const wrapper = shallow(<AttachmentImage {...baseProps}/>);
        expect(wrapper).toMatchSnapshot();
    });

    test('it sets state based on props', () => {
        const wrapper = shallow(<AttachmentImage {...baseProps}/>);

        const state = wrapper.state();
        expect(state.hasImage).toBe(true);
        expect(state.imageUri).toBe('https://images.com/image.png');
        expect(state.originalWidth).toBe(32);
    });

    test('it does not render image if no imageUrl is provided', () => {
        const props = {...baseProps, imageUrl: null, imageMetadata: null};
        const wrapper = shallow(<AttachmentImage {...props}/>);

        const state = wrapper.state();
        expect(state.hasImage).toBe(false);
        expect(state.imageUri).toBe(null);
    });

    test('it calls Image.getSize if metadata is not present', () => {
        const getSizeFn = jest.fn((_, callback) => {
            callback(64, 64);
        });
        Image.getSize = getSizeFn;

        const props = {...baseProps, imageMetadata: null};
        const wrapper = shallow(<AttachmentImage {...props}/>);

        const state = wrapper.state();
        expect(state.hasImage).toBe(true);
        expect(state.imageUri).toBe('https://images.com/image.png');
        expect(state.originalWidth).toBe(64);
        expect(getSizeFn).toHaveBeenCalled();
    });

    test('it updates image when imageUrl prop changes', () => {
        const wrapper = shallow(<AttachmentImage {...baseProps}/>);

        wrapper.setProps({
            imageUrl: 'https://someothersite.com/picture.png',
            imageMetadata: {
                width: 96,
                height: 96,
            },
        });

        const state = wrapper.state();
        expect(state.hasImage).toBe(true);
        expect(state.imageUri).toBe('https://someothersite.com/picture.png');
        expect(state.originalWidth).toBe(96);
    });

    test('it does not update image when an unrelated prop changes', () => {
        const wrapper = shallow(<AttachmentImage {...baseProps}/>);

        wrapper.setProps({
            theme: {...Preferences.THEMES.default},
        });

        const state = wrapper.state();
        expect(state.hasImage).toBe(true);
        expect(state.imageUri).toBe('https://images.com/image.png');
        expect(state.originalWidth).toBe(32);
    });
});
