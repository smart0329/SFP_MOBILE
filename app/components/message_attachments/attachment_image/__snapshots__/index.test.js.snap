// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AttachmentImage it matches snapshot 1`] = `
<TouchableWithFeedbackIOS
  onPress={[Function]}
  style={
    Array [
      Object {
        "marginTop": 5,
      },
      Object {
        "width": 38,
      },
    ]
  }
  type="none"
>
  <View
    style={
      Array [
        Object {
          "borderColor": "rgba(61,60,64,0.1)",
          "borderRadius": 2,
          "borderWidth": 1,
          "flex": 1,
        },
        Object {
          "height": 28,
          "width": 28,
        },
      ]
    }
  >
    <Connect(ProgressiveImage)
      imageStyle={
        Object {
          "marginBottom": 5,
          "marginLeft": 2.5,
          "marginRight": 5,
          "marginTop": 2.5,
        }
      }
      imageUri="https://images.com/image.png"
      resizeMode="contain"
      style={
        Object {
          "height": 28,
          "width": 28,
        }
      }
    />
  </View>
</TouchableWithFeedbackIOS>
`;
